src/azure_mail_reports.py

```diff
@@ -0,0 +1,87 @@
+import logging
+from dateutil import parser
+import re
+
+from datetime import datetime, timedelta
+from src.api import Api
+from src.data.oauth_api_data import OAuthApiData
+from src.oauth_api import OAuthApi
+
+logger = logging.getLogger(__name__)
+
+
+class AzureMailReports(OAuthApi):
+    DEFAULT_GRAPH_DATA_LINK = 'd.results'
+    AZURE_GRAPH_TOKEN_SCOPE = 'scope'
+    AZURE_GRAPH_FILTER_CONCAT = '&$'
```

Comment:
Since it's not azure graph, let's rename to something relevant to this type please